tests/com/tony/server/AuthenticatorTest.java

```diff
@@ -0,0 +1,74 @@
+package com.tony.server;
+
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+public class AuthenticatorTest {
+    @Test
+    public void testAddToAuthenticationList() throws Exception {
+        Authenticator authenticator = new Authenticator();
+        authenticator.addToAuthenticationList(new Request("GET", "/blah"));
+
+        assertTrue(authenticator.getAuthenticationList().contains(new Request("GET", "/blah")));
+    }
+
+    @Test
+    public void testAddToAuthenticatedUsers() throws Exception {
+        Authenticator authenticator = new Authenticator();
+        authenticator.addToAuthenticatedUsers("admin:hunter2");
+
+        assertTrue(authenticator.getAuthenticatedUsers().contains("admin:hunter2"));
+    }
+
+    @Test
+    public void testRequiresAuthenticationReturnsTrueIfIncludedInList() throws Exception {
+        Authenticator authenticator = new Authenticator();
+        authenticator.addToAuthenticationList(new Request("GET", "/blah"));
+
+        assertTrue(authenticator.requiresAuthentication(new Request("GET", "/blah")));
+    }
+
+    @Test
+    public void testRequiresAuthenticationReturnsFalseIfNotInList() throws Exception {
+        Authenticator authenticator = new Authenticator();
+        authenticator.addToAuthenticationList(new Request("GET", "/blah"));
+
+        assertFalse(authenticator.requiresAuthentication(new Request("GET", "/somethingElse")));
+    }
+
+    @Test
+    public void testIsAuthenticatedReturnsTrueIfMatchesUserInfo() throws Exception {
+        Authenticator authenticator = new Authenticator();
+        authenticator.addToAuthenticationList(new Request("GET", "/blah"));
+        authenticator.addToAuthenticatedUsers("admin:hunter2");
+
+        Request authenticatedRequest = new Request("GET", "/blah");
+        authenticatedRequest.addToHeaders("Authorization", "Basic YWRtaW46aHVudGVyMg==");
+        assertTrue(authenticator.isAuthenticated(authenticatedRequest));
+    }
+
+    @Test
+    public void testIsAuthenticatedReturnsFalseIfErrorOnDecode() throws Exception {
+        Authenticator authenticator = new Authenticator();
+        authenticator.addToAuthenticationList(new Request("GET", "/blah"));
+        authenticator.addToAuthenticatedUsers("admin:hunter2");
+
+        Request authenticatedRequest = new Request("GET", "/blah");
+        authenticatedRequest.addToHeaders("Authorization", "Basic WrongEncodingLength=123");
+        assertFalse(authenticator.isAuthenticated(authenticatedRequest));
+    }
+
+    @Test
+    public void testDecodeUserInfoReturnsCorrectDecoding() throws Exception {
+        assertEquals("admin:hunter2",
```

Comment:
I'm a little torn on this. This feels like an implementation detail, that [I don't think I would test directly](http://googletesting.blogspot.com/2013/08/testing-on-toilet-test-behavior-not.html) or even expose as a public method on the `Authenticator`. I'm also not sure this belongs in the `Authenticator`. What do you think?